sparsh2/CS-555-Senior-Sage
# Storage and Access Control services

This PR adds Storage service for RBAC and sets up helm charts for easy deployments of the project on K8s. This completes the JIRA: [SCRUM-66](https://team17seniorsage.atlassian.net/browse/SCRUM-66), [SCRUM-44](https://team17seniorsage.atlassian.net/browse/SCRUM-44), [SCRUM-65](https://team17seniorsage.atlassian.net/browse/SCRUM-65).

## How to test?
- Install minikube and helm (better to install with `chocolatey` package manager for windows)
	- Minikube: https://minikube.sigs.k8s.io/docs/start/?arch=%2Fwindows%2Fx86-64%2Fstable%2F.exe+download
	- Helm: https://helm.sh/docs/intro/install/
- Start minikube
	- `minikube start`
	- `minikube dashboard`
- Open `docker/Makefile` in editor. Change to user docker username at the top of the file
- In a separate terminal, build images
	- `cd docker`
	- `make`
- Install the helm package
	- `cd helm`
	- `helm install senior-sage ./senior-sage`
- In the minikube dashboard, change your namespace to `senior-sage` and navigate to `Pods`. You should see 2 pods - `authz` and `storage-service` pods

![image](https://github.com/user-attachments/assets/0c94de58-bc24-4aba-a8d7-68f21376262d)
